pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -389,3 +479,133 @@ func (n *clusterManagerController) cleanUp(
 
 	return nil
 }
+
+func cleanUpManagement(ctx context.Context, controllerContext factory.SyncContext,
+	kubeClient kubernetes.Interface, config manifests.HubConfig) error {
+	// Remove All Static files
+	managementResources := []string{namespaceResource}
+	for _, file := range managementResources {
+		err := helpers.CleanUpStaticObject(
+			ctx,
+			kubeClient, nil, nil,
+			func(name string) ([]byte, error) {
+				template, err := manifests.ClusterManagerManifestFiles.ReadFile(name)
+				if err != nil {
+					return nil, err
+				}
+				return assets.MustCreateAssetFromTemplate(name, template, config).Data, nil
+			},
+			file,
+		)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func generateHubClients(hubKubeConfig *rest.Config) (kubernetes.Interface, apiextensionsclient.Interface, apiregistrationclient.APIServicesGetter, error) {
+	hubClient, err := kubernetes.NewForConfig(hubKubeConfig)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	hubApiExtensionClient, err := apiextensionsclient.NewForConfig(hubKubeConfig)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	hubApiRegistrationClient, err := apiregistrationclient.NewForConfig(hubKubeConfig)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	return hubClient, hubApiExtensionClient, hubApiRegistrationClient, nil
+}
+
+func ensureSAKubeconfigs(ctx context.Context, clusterManagerName, clusterManagerNamespace string, clusterManagerMode operatorapiv1.InstallMode,
+	hubKubeConfig *rest.Config, hubClient, managementClient kubernetes.Interface, recorder events.Recorder) error {
+	// setup template kubeconfig
+	templateKubeconfig, err := getTemplateKubeconfig(hubKubeConfig, clusterManagerMode)
+	if err != nil {
+		return err
+	}
+	ensureSAToken := func(saName string) error {
+		return helpers.EnsureSAToken(ctx, saName, clusterManagerNamespace, hubClient,
+			helpers.RenderToKubeconfigSecret(saName+"-kubeconfig", clusterManagerNamespace, templateKubeconfig, managementClient.CoreV1(), recorder))
+	}
+	sas := getSAs(clusterManagerName)
+	err = retry.OnError(retry.DefaultBackoff, func(e error) bool {
+		return true
+	}, func() error {
+		for _, sa := range sas {
+			err = ensureSAToken(sa)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// getSAs return serviceaccount names of all hub components
+func getSAs(clusterManagerName string) []string {
+	return []string{
+		clusterManagerName + "-registration-controller-sa",
+		clusterManagerName + "-registration-webhook-sa",
+		clusterManagerName + "-work-webhook-sa",
+		clusterManagerName + "-placement-controller-sa",
+	}
+}
+
+func getTemplateKubeconfig(hubkubeconfig *rest.Config, mode operatorapiv1.InstallMode) (*rest.Config, error) {
+	switch mode {
+	case operatorapiv1.InstallModeDefault:
+		return &rest.Config{
+			Host: "https://kubernetes.default.svc",
```

Comment:
This is because we consider default mode as a special detached mode -- the management cluster and the hub cluster is one cluster. This aims to reduce the effort when we need to updates different modes.

This is also why we pass kubeconfig not a client into `helpers.GetHubKubeconfig`. Because in default mode, the `hubkubeconfi` is equal to `managementKubeconfig`.

Reply:
Fixed, using operatorKubeconfig(right now it must be InClusterConfig) in default mode.